CIP-1855/CIP-1855.md

```diff
@@ -0,0 +1,94 @@
+---
+CIP: 1855
+Title: Forging policy keys for HD Wallets
+Authors: Samuel Leathers <samuel.leathers@iohk.io>, John Lotoski <john.lotoski@iohk.io>, Michael Bishop <michael.bishop@iohk.io>
+Comments-Summary: Multi-party transaction signing and key management for HD wallets.
+Comments-URI: https://github.com/cardano-foundation/CIPs/wiki/Comments:CIP-1855
+Status: Draft
+Type: Standards
+Created: 2021-06-02
+License: CC-BY-4.0
+---
+
+# Abstract
+
+This document describes how to derive forging policy keys used for minting/burning tokens.
+
+## Glossary
+
+
+Term     | Definition
+---      | ---
+HD       | Hierarchical Deterministic, refers to wallets as described in [BIP-0032].
+
+# Motivation
+
+## Overview 
+
+Forging tokens is derived from a script policy. The script policy includes hashes of keys needed to forge new tokens and must be witnessed by the keys with hashes listed. 
+
+# Specification
+
+## HD Derivation
+
+We consider the following HD derivation paths similarly to [CIP-1852]:
+
+```
+m / purpose' / coin_type' / account_ix' / role / index
+```
+
+
+To associate policy keys to a wallet, we reserve however `purpose=1855'` to reserve for policy keys for forging tokens. The coin type remains `coin_type=1815'` to identify Ada as registered in [SLIP-0044]. We use a hardened index for each policy key as derivation is not needed.
+
+We can summarize the various paths and their respective domain in the following table:
+
+| `purpose` | `coin_type` | `policy_ix`        |
+| ---       | ---         | ---                |
+| `1855'`   | `1815'`     | `[2^31 .. 2^32-1]'` |
```

Comment:
I suppose this is where we start the bike shedding... :grin: 

Are there any downsides to instead using a new `role` index under `m / 1852´ / 1815´ / account´` (i.e. amendment to [CIP-1852](https://github.com/cardano-foundation/CIPs/blob/master/CIP-1852/CIP-1852.md) scheme)?

This might be easier for wallets to implement. The token policy keys would belong to a specific account of a wallet.

Also, is there a reason to choose a new `purpose` when this could be a new `coin_type` under Cardano `1852´`? We could use `coin_type` `2021´` in recognition of the very contemporary surge in popularity of NFTs.

Reply:
@rvl I would avoid changing the coin type because that requires a change to a SLIP to avoid clashing with other cryptocurrencies

RE: using a new role derivation instead
Is creating tokens something that we consider should be namespaced per-account or something specific to the CIP-1852 standard? Not sure if that's necessarily the case.

Reply:
Yes you are right - the token policy keys aren't necessarily directly related to a wallet account.